apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/RequestIdCallInterceptor.cs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Grpc.Core;
+using Grpc.Core.Interceptors;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Spanner.V1;
+
+/// <summary>
+/// A <see cref="Interceptor"/> that wraps all calls, adding the Spanner request ID to any exceptions thrown while
+/// using the <see cref="SpannerClient"/>.
+/// </summary>
+internal sealed class RequestIdCallInterceptor : Interceptor
```

Comment:
I think we can do with a singleton of this class, make the default constructor private and add a static `Instace` property.